lib/ansible/module_utils/_text.py

```diff
@@ -1,278 +1,9 @@
-# This code is part of Ansible, but is an independent component.
-# This particular file snippet, and this file snippet only, is BSD licensed.
-# Modules you write using this snippet, which is embedded dynamically by Ansible
-# still belong to the author of the module, and may assign their own license
-# to the complete work.
-#
-# Copyright (c), Toshio Kuratomi <a.badger@gmail.com>, 2016
-#
-# Redistribution and use in source and binary forms, with or without modification,
-# are permitted provided that the following conditions are met:
-#
-#    * Redistributions of source code must retain the above copyright
-#      notice, this list of conditions and the following disclaimer.
-#    * Redistributions in binary form must reproduce the above copyright notice,
-#      this list of conditions and the following disclaimer in the documentation
-#      and/or other materials provided with the distribution.
-#
-# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
-# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
-# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
-# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
-# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
-# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
-# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-#
+# Copyright (c), Toshio Kuratomi <tkuratomi@ansible.com> 2016
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+# Backwards compat for people still calling it from this package
+from ansible.module_utils.common.text.converters import to_bytes, to_native, to_text
 
 """
-.. warn:: This module_util is currently internal implementation.
-    We want to evaluate this code for stability and API suitability before
-    making backwards compatibility guarantees.  The API may change between
-    releases.  Do not use this unless you are willing to port your module code.
+.. warn:: Use ansible.module_utils.common.text.converters instead.
 """
```

Comment:
We already have a warnings mechanism in `module_utils.common.warnings` that is consumed on the module side, but we need to implement consuming of these warnings controller side.  As such, it's not a completed feature yet.

Reply:
Well, the built-in Python warnings mechanism is probably better in this case because the emitted warnings can be automatically identified by pytest and are controlled by `python -W`. That `module_utils.common.warnings` only collects lists of strings to print out, external systems have no way of knowing that those strings are in fact warnings which is mostly fine for the remote execution because we control it.

Reply:
Alright, after seeing the messages on slack I think that adding warnings can wait but still the docstring must be relocated because it was a module docstring originally and we should keep it on the top of the module.
```suggestion

"""
.. warn:: Use ansible.module_utils.common.text.converters instead.
"""

# Backwards compat for people still calling it from this package
from ansible.module_utils.common.text.converters import to_bytes, to_native, to_text
```

Reply:
I would not add a runtime warning so soon after adding the new location, especially for something like this which is used in so many places in the Ansible codebase. I also don't believe we use the stdlib warnings mechanism for our warning setup so this won't actually do anything for Ansible right now.

That could change in the future but we should at least have this in place for 1 release before annoying users to move away from it.